code generators/de.fhdo.lemma.model_processing.code_generation.springcloud.zuul/src/main/java/de/fhdo/lemma/model_processing/code_generation/springcloud/zuul/template/ZuulTemplate.xtend

```diff
@@ -0,0 +1,251 @@
+package de.fhdo.lemma.model_processing.code_generation.springcloud.zuul.template
+
+import de.fhdo.lemma.operation.intermediate.IntermediateInfrastructureNode
+import java.io.File
+
+/**
+ * Template class for generating operation node specific components.
+ *
+ * @author <a href="mailto:philip.wizenty@fh-dortmund.de">Philip Wizenty</a>
+ */
+class ZuulTemplate {
+    /**
+     * Kubernetes Deployment template for generating a Zuul deployment.
+     */
+    static def String getKubernetesDeploymentForZuul(IntermediateInfrastructureNode node)'''
+        apiVersion: apps/v1
+        kind: Deployment
+        metadata:
+          creationTimestamp: null
+          labels:
+            app: «node.name.toLowerCase»
+          name: «node.name.toLowerCase»
+        spec:
+          replicas: 1
+          selector:
+            matchLabels:
+              app: «node.name.toLowerCase»
+          strategy: {}
+          template:
+            metadata:
+              creationTimestamp: null
+              labels:
+                app: «node.name.toLowerCase»
+            spec:
+              containers:
+              - image: «node.name.toLowerCase»
+                imagePullPolicy: "Never"
+                name: «node.name.toLowerCase»
+                resources: {}
+        status: {}
+        ---
+        apiVersion: v1
+        kind: Service
+        metadata:
+          creationTimestamp: null
+          labels:
+            app: «node.name.toLowerCase»
+          name: «node.name.toLowerCase»
+        spec:
+          ports:
+          «FOR property : node.defaultValues»
+            «IF(property.technologySpecificProperty.name.toLowerCase == "port")»
+                  - name: "«property.value»"
+                    port: «property.value»
+                    targetPort: «property.value»
+            «ENDIF»
+          «ENDFOR»
+          selector:
+            app: «node.name.toLowerCase»
+          type: ClusterIP
+        status:
+          loadBalancer: {}
+    '''
+
+    /**
+     * Create Docker-Compose component for Zuul
+     */
+    static def String getDockerComposeForZuul(IntermediateInfrastructureNode node)'''
+        «node.name.toLowerCase»:
+          image: «node.name.toLowerCase»
+          build: «node.name.toLowerCase + File.separator»
+          container_name: «node.name.toLowerCase»
+          «FOR property : node.defaultValues»
+            «IF(property.technologySpecificProperty.name.toLowerCase == "port")»
+                ports:
+                  - "«property.value»:«property.value»"
+          «ENDIF»
+        «ENDFOR»
+          networks:
+            - default-network
+    '''
+
+    /**
+     * Create a Spring-based application main class for Zuul in Kotlin
```

Comment:
Why Kotlin and not Java at all? Currently, the Java Base Generator also supports only Java. Is the motivation for Kotlin related to Zuul?

Reply:
If not, please create an issue (to the least) to enable alternative generation of a Java-based Zuul service. This approach would comply to the Java Base Generator, which provides sophisticated support for Java (and only Java) as a microservice programming language.

Reply:
There is no Zuul related motivation for the usage of Kotlin as a programming language. 
I just liked Kotlin more at that moment. 

I will add the support for Java-based Zuul Application gateways, to also support the Java programming language, that is used in the Java Base generator.

Reply:
Yeah, or just create an issue for now.